### PR TITLE
Implemented replaceTextWithScreenVariables function, added MarkdownVa…

### PR DESCRIPTION
**Pull request by vmkhitaryanscn**
…riableReplacer class, added unit tests for these

### CODE REVIEW

File: src/shared/lib/utils/survey.ts [hunk 1]
```diff
@@ -13,4 +15,208 @@ export const invertColor = (hex: string) => {
...
+
+type ActivityItems = {
+  variableName: string;
+  inputType: InputAnswer['inputType'];
+  valueConstraints: {
+    itemList: { name: string; value: string | number }[];
+  };
+}[];
+
+export class MarkdownVariableReplacer {
```
Comment (BamMironov) [inline src/shared/lib/utils/survey.ts:100]: Please put it in a separate file.
You can create the "src/shared/survey" folder and move there everything that you created here.

File: src/shared/lib/utils/survey.ts [hunk 2]
```diff
@@ -13,4 +15,208 @@ export const invertColor = (hex: string) => {
...
+  private markdown = '';
```
Comment (BamMironov) [inline src/shared/lib/utils/survey.ts:101]: I think this should not be private property.

The `MarkdownVariableReplacer` must accept two arguments: `activityItems` and `answers`.
Then, the `process` function must accept one argument: `markdown`.
This way, we can create a replacer based on a context (activityItems, answers) and then process any markdown with it.

File: src/shared/lib/utils/survey.ts [hunk 3]
```diff
@@ -13,4 +15,208 @@ export const invertColor = (hex: string) => {
...
+    );
+    const answerNotFound = foundIndex < 0 || !this.answers[foundIndex];
+
+    if (answerNotFound) {
+      return `[[${variableName}]]`;
+    }
+
+    const activityItem = this.activityItems[foundIndex];
+    let updated = '';
+    const inputAnswer: InputAnswer = <InputAnswer>{
```
Comment (BamMironov) [inline src/shared/lib/utils/survey.ts:183]: Is it necessary to put <InputAnswer> generic here?

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/shared/lib/utils/survey.ts (hunk 3 above)
Comment (vmkhitaryanscn) [reply src/shared/lib/utils/survey.ts:183]: @BamMironov yes, it gives

TS2322: Type '{ inputType: "timeRange" | "date" | "text" | "radio" | "ageSelector" | "slider" | "checkbox"; answer: Answer; }' is not assignable to type 'InputAnswer'.   Type '{ inputType: "timeRange" | "date" | "text" | "radio" | "ageSelector" | "slider" | "checkbox"; answer: Answer; }' is not assignable to type 'CheckBoxInput'.     Types of property 'inputType' are incompatible.       Type '"timeRange" | "date" | "text" | "radio" | "ageSelector" | "slider" | "checkbox"' is not assignable to type '"checkbox"'.         Type '"timeRange"' is not assignable to type '"checkbox"

**Review by BamMironov — APPROVED** [review]

**Pull request merged**